### PR TITLE
Fixed JS file to work on new servers

### PR DESCRIPTION
**Pull request by DannyMurphy**

### CODE REVIEW

**Comment by kehh** [comment]
Hi Danny,
I'm not sure how this will play if there are existing values in the field (the _new bits in those attribute id's indicate that that is only for new values, so you may have to add selectors for if values have been inserted in the database and then somebody changes the radio button after the fact).

Cheers,
K

File: js/local/local.js [hunk 1]
```diff
@@ -169,6 +170,80 @@ wam.initConditionRecommendationTypes = function() {
 			}
 		});
 	}
+  
+    //Live
+    if ($('#P526OccurrenceEditorForm_attribute_318Item_new_0').length) {
```
Comment (leftclickben) [inline js/local/local.js:175]: I would recommend moving these identifier selectors into a configuration object, which could just be at the top of this script:

```
var selectors = {
    occurrenceEditorForm: {
        fieldName: '#SomeReallyLong_andStupid_lookingSelector_1_2_3' //, etc
    }
};
```

Then this line would become something like:

```
if ($(selectors.occurrenceEditorForm.whateverFieldThisIsFor).length) {
```

There are a few other selectors below as well (`wamcon`) so they could probably go into config as well.

The advantage of this is that you put all the selectors in one place, easy to update, and also are decoupling the implementation of the script from the particular DOM elements you happen to be manipulating.

Another advantage is that you are giving the selectors meaningful names, so you know what field it is ("attribute 318" means nothing).

**Comment by leftclickben** [comment]
Re @kehh comment about `_new` only working for new items, it also has `_0` after that, so it will also only work for the _first_ new item added.  Even if it this metadata element has a configured maximum of 1 value, check what happens if you add a record, then remove it, then add a new one.

**Comment by DannyMurphy** [comment]
Cool can we put a custom attribute on a user generated form element? That would be far more reusable and efficient.

D

**Comment by kehh** [comment]
ah sorry confusion of attributes - attribute in CA ~== a field in Drupal. I definitely want arbitrary html attributes to be specifyable on the form (or at least a rule that adds a class like 'attribute_value_{attributeCode}', but it doesn't do that right now).

**Pull request closed without merging**